llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -13457,6 +13457,55 @@ combineBinOp_VLToVWBinOp_VL(SDNode *N, TargetLowering::DAGCombinerInfo &DCI) {
   return InputRootReplacement;
 }
 
+// (vwadd y, (select cond, x, 0)) -> select cond (vwadd y, x), y
+static SDValue combineVWADDSelect(SDNode *N, SelectionDAG &DAG) {
+  unsigned Opc = N->getOpcode();
+  assert(Opc == RISCVISD::VWADD_VL || Opc == RISCVISD::VWADD_W_VL ||
```

Comment:
It can't be VWADD_VL due to the check in `performVWADD_VLCombine` right?

Reply:
The check in `performVWADD_VLCombine` is for  `RISCVISD::VWADD_W_VL` and `RISCVISD::VWADDU_W_VL`. 
We need to first do `combineBinOp_VLToVWBinOp_VL` on those.

Reply:
Oops. You're right. Sorry about that.